homeassistant/components/sensor/__init__.py

```diff
@@ -546,6 +546,81 @@ class SensorStateClass(StrEnum):
     SensorDeviceClass.WIND_SPEED: set(UnitOfSpeed),
 }
 
+DEVICE_CLASS_STATE_CLASSES: dict[SensorDeviceClass, set[SensorStateClass | None]] = {
+    SensorDeviceClass.APPARENT_POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.AQI: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ATMOSPHERIC_PRESSURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.BATTERY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CO: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CO2: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.CURRENT: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_RATE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.DATA_SIZE: {
+        None,
+        SensorStateClass.MEASUREMENT,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.DATE: {None},
+    SensorDeviceClass.DISTANCE: {None}.union(set(SensorStateClass)),
+    SensorDeviceClass.DURATION: {None},
+    SensorDeviceClass.ENERGY: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.ENUM: {None},
+    SensorDeviceClass.FREQUENCY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.GAS: {
+        None,
+        SensorStateClass.TOTAL,
+        SensorStateClass.TOTAL_INCREASING,
+    },
+    SensorDeviceClass.HUMIDITY: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.ILLUMINANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.IRRADIANCE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MOISTURE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.MONETARY: {None, SensorStateClass.TOTAL},
+    SensorDeviceClass.NITROGEN_DIOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROGEN_MONOXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.NITROUS_OXIDE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.OZONE: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM1: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM10: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PM25: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER_FACTOR: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.POWER: {None, SensorStateClass.MEASUREMENT},
+    SensorDeviceClass.PRECIPITATION: {
+        None,
+        SensorStateClass.MEASUREMENT,
```

Comment:
Why can this be a measurement? Measuring instant precipitation (mm/sec?) doesn't make much sense.

Reply:
This is participation depth, not intensity

Reply:
Yes, sorry if that wasn't expressed well. Measurement makes sense for the rate but not for the depth.

Reply:
Hmm I think I see what you mean. But then there is still an inconsistency, and `measurement` should also be valid to volume and weight.

Reply:
Oh I agree, I originally did it but removed it because of a comment from @epenet : <https://github.com/home-assistant/core/pull/84402#discussion_r1055277693>

Reply:
Yes I saw that and that's why I thought of distance. So are you putting it back? :)

Reply:
I do not know or care for this particular device class. It is an odd one, for that reason, I added them all (which is effectively the same as we currently have). This PR does therefore not enforce or change behavior for this device class.

Reply:
Yeah I understand. I am talking of volume/weight now, which are inconsistent with others such as distance or precipitation. I don't care if measurement should be included or not but they should be the same.

Reply:
@emontnemery I think you know best what to do in this case?

Reply:
Discussed this with Erik, it is an accumulation, and should thus not have a measurement state class. Erik opens up docs to make clear it is an accumulation.

../Frenck